src/iso19111/c_api.cpp

```diff
@@ -624,6 +624,69 @@ int proj_uom_get_info_from_database(PJ_CONTEXT *ctx, const char *auth_name,
     return false;
 }
 
+
+// ---------------------------------------------------------------------------
+
+/** \brief Get information for a grid from a database lookup.
+ *
+ * @param ctx Context, or NULL for default context.
+ * @param grid_name Grid name (must not be NULL)
+ * @param out_full_name Pointer to a string value to store the grid full
+ * filename. or NULL
+ * @param out_package_name Pointer to a string value to store the package name
+ * where
+ * the grid might be found. or NULL
+ * @param out_url Pointer to a string value to store the grid URL or the
+ * package URL where the grid might be found. or NULL
+ * @param out_direct_download Pointer to a int (boolean) value to store whether
+ * *out_url can be downloaded directly. or NULL
+ * @param out_open_license Pointer to a int (boolean) value to store whether
+ * the grid is released with an open license. or NULL
+ * @param out_available Pointer to a int (boolean) value to store whether the
+ * grid is available at runtime. or NULL
+ * @return TRUE in case of success.
+ */
+int PROJ_DLL proj_grid_get_info_from_database(PJ_CONTEXT *ctx,
+                               const char *grid_name,
+                               const char **out_full_name,
+                               const char **out_package_name,
+                               const char **out_url,
+                               int *out_direct_download,
+                               int *out_open_license,
+                               int *out_available) {
+    assert(grid_name);
+    SANITIZE_CTX(ctx);
+    try {
+        auto db_context = getDBcontext(ctx);
+        std::string full_name_str;
+        std::string package_name_str;
+        std::string url_str;
+        bool direct_download;
+        bool open_license;
+        bool available;
+        if ( !db_context->lookForGridInfo( grid_name, full_name_str, package_name_str, url_str, direct_download, open_license, available ) )
+            return false;
+
+        if ( out_full_name )
+            *out_full_name = full_name_str.c_str();
```

Comment:
unfortunately the C pointer will no longer be valid once that function returns, since full_name_str is a temporary object. Same for package_name and url. You probably need to store those objects in the ctx class.